dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -3209,8 +3194,12 @@ private Contentlet checkin(Contentlet contentlet, ContentletRelationships conten
 
                             String velocityVarNm = field.getVelocityVarName();
                             java.io.File incomingFile = contentletRaw.getBinary(velocityVarNm);
+                            if(incomingFile.length()==0 && !Config.getBooleanProperty("CONTENT_ALLOW_ZERO_LENGTH_FILES", false)){
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)